contracts/contracts/vault/VaultCore.sol

```diff
@@ -53,16 +52,10 @@ contract VaultCore is VaultStorage {
                 .scaleBy(int8(10)), // 18-8 because oracles have 8 decimals precision
             10**Helpers.getDecimals(_asset)
         );
-        if (
-            (priceAdjustedDeposit > rebaseThreshold && !rebasePaused) ||
-            (priceAdjustedDeposit >= autoAllocateThreshold)
-        ) {
-            assetPrices = _getAssetPrices(false);
-        }
 
         // Rebase must happen before any transfers occur.
         if (priceAdjustedDeposit > rebaseThreshold && !rebasePaused) {
-            rebase(assetPrices, true);
+            rebase(true);
```

Comment:
maybe a nit - bool params are icky - perhaps better to make rebaseWithSync() and rebaseWithoutSync - and in any case - why need to to include the sync on the pairs - anyway whoever does the first interaction with the pair will cause a sync , so could avoid a cost to that call to the pairs

Reply:
Not sure I understand, the sync needs to be called on every Uniswap pair every time the rebase is called because it changes the balances held in Uniswap which breaks the constant product requirement

Reply:
The pair itself will call sync anytime anyone interacts with it - like add liquidity/remove liquidity or a swap. So its isn't strictly necessary for us to call sync , since sync anyway will be called. The pair itself will check balances of its tokens

Reply:
It'll call sync after a swap (for example), but the transaction will revert prior to that if the balances are out of sync https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2Pair.sol#L182

Reply:
what you listed was making sure that the LP fee was taken into account - for example - after the yam rebase occured (the one that caused a massive amount of new token) , people were still able to do a swap - and that initial swap caused a sync (that first person got a great deal) 

Let me try to give an example wrt this PR

Reply:
I can't see the fee used in that swap function. Per the error `UniswapV2: K`, k is the constant product.

Reply:
Lets say I run an apple store during these times of coronavirus. Only one customer at time in my store, and I only go into the store after each customer to check on things. I also have a time machine that can revert. 

So a customer walks into the store, takes some apples, leaves some cash. I walk into the store, carrying my accounting books. I count up the cash in the store, the number of apples in the store, compare both to my books, and if the customer left too little money I use the time machine to revert. Otherwise, I update my books and walk back out.

This means that I don't use the time machine if a customer left too much money, or took too few apples. Importantly, this means that if the apples in the store spontaneously replicate, than a customer can take those extra apples out with a normal order, and I won't know the difference.

In our case, calling sync provides two protections. First, if a horse snuck into the store and ate some apples, then the next customer could walk in, do a perfectly fair transaction, and get reverted. Syncing stops this. Secondly, the sync prevents someone who knows when there are extra apples from gradually gaining a greater and greater share of the  at the expense of everyone else.